frontend/src/stores/NewsStore.js

```diff
@@ -0,0 +1,105 @@
+import { Store } from 'flux/utils';
+import ApiService from '@Services/ApiService.js';
+import AppDispatcher from '@Services/AppDispatcher.js';
+import { NewsActions, updateNewsAction, updateArticleAction, updateArticleDeletedAction, addArticleAction } from '@Actions/NewsActions.js';
+
+const urls = {
+  createNews: 'news',
+  readNews: 'news',
+  updateNews: 'news/update',
+  deleteNews: 'news'
+}
+
+/**
+ * Stores and updates all news articles
+ */
+class NewsStore extends Store {
+  constructor(dispatcher) {
+    super(dispatcher);
+
+    this.news = [];
+
+    // possible errors
+    this.error = undefined;
+  }
+
+  /**
+   * Override of base class that is called when handling actions. Remember to not
+   * dispatch new actions in this function.
+   * @param {object} payload
+   */
+  __onDispatch(payload) {
+    switch (payload.action) {
+      case NewsActions.GET_NEWS:
+        AppDispatcher.dispatchPromisedFn(
+          ApiService.readAuthenticatedData(urls.readNews, {}),
+          data => updateNewsAction(data),
+          (errData) => {
+            this.error = errData;
+          },
+        );
+        break;
+      case NewsActions.UPDATE_NEWS:
+        this.news = payload.news;
+        this.__emitChange();
+        break;
+      case NewsActions.CREATE_NEWS:
+        AppDispatcher.dispatchPromisedFn(
+          ApiService.createData(urls.createNews, {
+            title: payload.news.title,
+            content: payload.news.content
+          }),
+          data => addArticleAction(data),
+          (errData) => {
+            this.error = errData;
+          }
+        );
+        break;
+      case NewsActions.EDIT_NEWS:
+        AppDispatcher.dispatchPromisedFn(
+          ApiService.updateData(urls.updateNews, {
+            id: payload.id,
+            title: payload.news.title,
+            content: payload.news.content
+          }),
+          data => updateArticleAction(data),
+          (errData) => {
+            this.error = errData;
+          },
+        );
+        break;
+      case NewsActions.DELETE_NEWS:
+        AppDispatcher.dispatchPromisedFn(
+          ApiService.deleteData(urls.deleteNews + '/' + payload.id, {}),
+          () => updateArticleDeletedAction(payload.id),
+          (errData) => {
+            this.error = errData;
```

Comment:
Denk dat hier nog een this.__emitChange() moet? Same bij bovestaande error handlers

Reply:
Er gebeurt momenteel eigenlijk niets met die errors. Moet eigenlijk wel, dus dat ga ik nu maar doen :-)

Reply:
Done.